### PR TITLE
[BREAKING][config] refactor: modify args structure

### PR DESCRIPTION
**Pull request by FoolPlayer**
### What does this PR do?

modify args structure

### CODE REVIEW

File: veomni/arguments/arguments_types.py [hunk 1]
```diff
@@ -553,180 +390,405 @@ class TrainingArguments:
...
+# ================================ Model Arguments ======================================
+#
+# Hierarchy:
+#   model.*
+#   └── network.*            → NetworkConfig
+#
+
+
+@dataclass
+class NetworkConfig:
```
Comment (Coach257) [inline veomni/arguments/arguments_types.py]: `Network` sounds strange. Maybe `ops_implementations`

File: configs/dit/flux_sft.yaml [hunk 2]
```diff
@@ -4,56 +4,64 @@ model:
...
   height: 1024
   width: 768
   datasets_repeat: 10
 
 train:
-  output_dir: Flux_SFT
-  train_architecture: full
-  data_parallel_mode: fsdp1
-  ulysses_parallel_size: 1
+  accelerator:
```
Comment (Luosuu) [inline configs/dit/flux_sft.yaml:28]: do we need this level?

File: configs/dit/flux_sft.yaml [hunk 3]
```diff
@@ -4,56 +4,64 @@ model:
   pretrained_text_encoder_2_path: FLUX.1-dev/text_encoder_2
   pretrained_vae_path: FLUX.1-dev/ae.safetensors
   config_path: ./configs/model_configs/flux/flux.json
-  attn_implementation: flash_attention_2
+  network:
```
Comment (Luosuu) [inline configs/dit/flux_sft.yaml]: do we need this level?

File: configs/dit/flux_sft.yaml (hunk 2 above)
Comment (FoolPlayer) [reply configs/dit/flux_sft.yaml:28]: This level is for settings like parallal or offload for improving the training.

File: configs/dit/flux_sft.yaml (hunk 3 above)
Comment (FoolPlayer) [reply configs/dit/flux_sft.yaml]: Maybe change to `ops_implementations` ?

File: veomni/arguments/arguments_types.py [hunk 4]
```diff
@@ -25,14 +25,97 @@
 logger = logging.get_logger(__name__)
 
 
-# ================================ Sub Training Arguments ======================================
+# ================================ Training Arguments ======================================
```
Comment (piyifan123) [inline veomni/arguments/arguments_types.py:28]: nit: should we mark this PR as breaking since it breaks existing training configs?

File: .github/workflows/gpu_e2e_test.yml [hunk 5]
```diff
@@ -86,7 +86,7 @@ jobs:
 
   cleanup:
```
Comment (piyifan123) [inline .github/workflows/gpu_e2e_test.yml:87]: since it's really a large change, would you mind using an AI to summarize the changes into the PR description so it's easy to see if this MR is just a renaming?

**Review by piyifan123 — APPROVED** [review]
LGTM

did not review details as it's mostly no-op renaming changes.

File: .github/workflows/gpu_e2e_test.yml (hunk 5 above)
Comment (FoolPlayer) [reply .github/workflows/gpu_e2e_test.yml:87]: Sure!

**Comment by FoolPlayer** [comment]
Here is the diff.

## Refactor: Hierarchical Argument Structure

Reorganize `VeOmniArguments` from a flat structure into a nested, multi-level hierarchy for better readability and maintainability. All existing functionality is preserved; only argument access paths change.

### New Argument Hierarchy

```
model.*
├── ops_implementation.*          (attn_implementation, moe_implementation)

train.*
├── optimizer.*                   (type, lr, weight_decay, warmup, scheduler, ...)
├── wandb.*                       (enable, project, run_name, ...)
├── profile.*                     (enable, rank0_only, ...)
├── gradient_checkpointing.*      (enable, debug, enable_reentrant)
├── accelerator.*                 (dp_*_size, tp_size, ep_size, ulysses_size, ...)
│   ├── fsdp_config.*             (fsdp_mode, full_shard, reshard_after_forward, ...)
│   └── offload_config.*          (enable_activation, activation_gpu_limit)
└── checkpoint.*                  (output_dir, load_path, manager, ...)

data.*
└── dataloader.*                  (dataloader_type, num_workers, prefetch_factor, ...)
```


### Key Migration Reference

| Before | After |
|---|---|
| `--train.use_wandb` | `--train.wandb.enable` |
| `--train.wandb_project` | `--train.wandb.project` |
| `--train.enable_gradient_checkpointing` | `--train.gradient_checkpointing.enable` |
| `--train.dp_mode` | `--train.accelerator.fsdp_config.fsdp_mode` |
| `--train.tp_size` | `--train.accelerator.tp_size` |
| `--train.ep_size` | `--train.accelerator.ep_size` |
| `--train.full_shard` | `--train.accelerator.fsdp_config.full_shard` |
| `--train.activation_offload` | `--train.accelerator.offload_config.enable_activation` |
| `--train.activation_gpu_limit` | `--train.accelerator.offload_config.activation_gpu_limit` |
| `--train.output_dir` | `--train.checkpoint.output_dir` |
| `--train.lr` | `--train.optimizer.lr` |
| `--train.weight_decay` | `--train.optimizer.weight_decay` |
| `--model.attn_implementation` | `--model.ops_implementation.attn_implementation` |
| `--model.moe_implementation` | `--model.ops_implementation.moe_implementation` |
| `--data.dataloader_type` | `--data.dataloader.dataloader_type` |

### Files Changed

- `veomni/arguments/arguments_types.py` — core restructure with new sub-config dataclasses
- `veomni/arguments/__init__.py` — updated exports
- `configs/**/*.yaml` (19 files) — updated YAML keys to match new hierarchy
- `tasks/**/*.py` (5 files) — updated argument access paths
- `tests/**/*.py` (13 files) — updated test fixtures and CLI args
- `docs/usage/arguments.md` — redesigned as structured API reference

**Comment by Coach257** [comment]
https://github.com/ByteDance-Seed/VeOmni/blob/29d396a2b93d95eda539407b406df3cc85d712a2/veomni/arguments/arguments_types.py#L516
Mv this to ckpt.__post__init__
https://github.com/ByteDance-Seed/VeOmni/blob/29d396a2b93d95eda539407b406df3cc85d712a2/veomni/arguments/arguments_types.py#L435
Mv this to acc.__post__init__
@FoolPlayer
